src/components/FeatureViews/Registration/RegistrationCart/StudentRegistrationsEntry.js

```diff
@@ -14,27 +14,37 @@ import TextField from "@material-ui/core/TextField";
 import CloseIcon from "@material-ui/icons/Close";
 import "./RegistrationCart.scss";
 import IconButton from "@material-ui/core/IconButton";
-import {removeRegistration, useValidateRegisteringParent} from "../../../OmouComponents/RegistrationUtils";
+import {useValidateRegisteringParent} from "../../../OmouComponents/RegistrationUtils";
+import {useDispatch} from "react-redux";
+import * as types from "../../../../actions/actionTypes";
+import NoListAlert from "../../../OmouComponents/NoListAlert";
 
 const separator = () => {
-	return (<Typography style={{ display: "inline", paddingLeft: 30, paddingRight: 30 }}>|</Typography>)
+	return (<Typography style={{display: "inline", paddingLeft: 30, paddingRight: 30}}>|</Typography>)
 }
 
-function RegistrationEntry({ registration: { course, numSessions, checked }, studentId }) {
+function RegistrationEntry({registration: {course, numSessions, checked}, studentId, index}) {
 	const {parentIsLoggedIn} = useValidateRegisteringParent();
 	const {updateSession} = useContext(RegistrationContext);
+	const dispatch = useDispatch();
 
 	const handleSessionChange = (e) => {
 		updateSession(Number(e.target.value), checked, studentId, course.id);
 	};
 
 	const handleSessionCheckChange = (_) => {
 		updateSession(numSessions, !checked, studentId, course.id);
-		removeRegistration(studentId, course.id);
+		dispatch({
+			type: types.DELETE_COURSE_REGISTRATION,
+			payload: {
+				studentId,
+				courseId: course.id,
+			}
+		});
```

Comment:
We should be migrating this to graphQL at some point.

Reply:
This isn't to delete a course registration in the database. it deletes it from the parent's registration cart which is why i need to use redux to do this